addons/mail/static/src/discuss/call/common/call_settings.js

```diff
@@ -65,6 +86,39 @@ export class CallSettings extends Component {
         return isMobileOS();
     }
 
+    async testMicrophoneVolume() {
+        if (!this.state.isTestMicrophoneReady) {
+            return;
+        }
+        this.state.isTestMicrophoneReady = false;
+        this.disconnectAudioMonitor?.();
+        this.disconnectAudioMonitor = undefined;
+        if (this.audioTrack) {
+            this.audioTrack.stop();
+            this.audioTrack = null;
+            this.state.isTestMicrophoneReady = true;
+            this.state.volume = 0;
+            return;
+        }
+        const audioStream = await browser.navigator.mediaDevices.getUserMedia({
+            audio: this.store.settings.audioConstraints,
+        });
+        const track = audioStream.getAudioTracks()[0];
+        if (this.isClosed) {
+            track.stop();
+            return;
+        }
+        this.audioMonitorPromise = monitorAudio(track, {
```

Comment:
Instead of `this.isClosed`, couldn't you check `owl.status()` instead?

Reply:
I didn't know it existed, when is it set to `"destroyed"`? I think that there could be some race if the promise of the track is resolved right between the stack of `willUnmount` and the status change to `"destroyed"`.